src/index.ts

```diff
@@ -0,0 +1,31 @@
+import type { App, Plugin } from 'vue'
+
+import QrcodeStream from './components/QrcodeStream.vue'
+import QrcodeCapture from './components/QrcodeCapture.vue'
+import QrcodeDropZone from './components/QrcodeDropZone.vue'
+
+// Install the components
+export function install(app: App) {
+  app.component('qrcode-stream', QrcodeStream)
+  app.component('qrcode-capture', QrcodeCapture)
+  app.component('qrcode-drop-zone', QrcodeDropZone)
+}
+
+// Expose the components
+export { QrcodeStream, QrcodeCapture, QrcodeDropZone }
+
+// Plugin definition
+const plugin: Plugin = { install }
+
+export { plugin as VueQrcodeReader }
+
+// // Auto-install
+// let GlobalVue = null
+// if (typeof window !== 'undefined') {
+//   GlobalVue = window.Vue
+// } else if (typeof global !== 'undefined') {
+//   GlobalVue = global.Vue
+// }
+// if (GlobalVue) {
+//   GlobalVue.use(plugin)
+// }
```

Comment:
I suspected those statements to always bundle vue-qrcode-reader, but I'm new to vue plugin auto imports, so I'm not sure how they work rn. Also, as you might expect, typescript complained :wink:

Reply:
I mean I didn’t came up with this. This was part of the template plugin, this project is based on. Could very well be that the best practice has changed.

I think the point of this is for non-npm users. So if you include the package just via script tag, then it’s „installed“ immediately. Probably not a big deal, since we are introducing breaking changes anyway. But then the install instructions in the README should probably be adjusted.

Reply:
Reverted those changes